src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -53,16 +54,61 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
                     continue;
                 }
 
-                if (node.TagHelper.IsBindTagHelper() && node.AttributeName.StartsWith("bind"))
+                if (node.TagHelper.IsBindTagHelper() && node.AttributeName.StartsWith("bind") && !node.IsParameterMatch)
```

Comment:
I realized I may have made this method a little more complicated that I should have. I'll try to simplify it further or add comments.

Reply:
Should we look at this or wait for simplification?

Reply:
You should still look. At least to check if it is actually complicated or makes sense.

Reply:
I've added some more comments.